core/src/main/java/bisq/core/btc/TxFeeEstimationService.java

```diff
@@ -58,7 +58,7 @@
     public static int TYPICAL_TX_WITH_1_INPUT_VSIZE = 175;
     private static int DEPOSIT_TX_VSIZE = 233;
 
-    private static int BSQ_INPUT_INCREASE = 150;
+    private static int BSQ_INPUT_INCREASE = 65;
```

Comment:
I think this might be too low. I just had a look at this and the table I came up with was, to be on the safe side

Tx size:
Element|size (vbyte)
--|--
Overhead|10
Inputs|41 * N
Outputs|31 * N_sw + 34 * N_p2pkh
Sig p2pkh|108
Sig SW|29

So I think the BSQ sw input should be counted as 41 + 29 = 70 vbytes extra, although it's sometimes a bit smaller.

Reply:
70 sounds good.  Do you have any thoughts on the caveat (legacy BSQ utxo) described above?

Reply:
For the atomic protocol I use this code to calculate the size. It overestimates a little bit due to sigs sometimes being a byte shorter, but otherwise quite accurate I think. This works since each side is only paying for their own and known inputs and outputs. Not sure if this will work for the normal trade protocol.

```
        // Overhead
        var size = payForOverhead ? 10L : 0L;

        // Inputs and signatures
        for (var input : parent.tx.getInputs()) {
            size += 41 + (input.hasWitness() ? 29 : 108);
        }

        // Outputs
        for (var output : parent.tx.getOutputs()) {
            size += ScriptPattern.isP2WPKH(output.getScriptPubKey()) ? 31 : 34;
        }

        return txFeePerVbyte.multiply(size);
```

Reply:
Unfortunately the Bisq code which does fee estimation has no visibility into the utxo being used.  BitcoinJ is used for that and it also has no visibility into the BSQ side of things, hence that constant `BSQ_INPUT_INCREASE`.

Reply:
Indeed, so for now I think this is fine. It's also the taker that pays for the whole tx which makes even worse to get an accurate calculation.

Perhaps we should wait with this change and add in the next release when more BSQ utxos are segwit. If included right now it would guarantee to be wrong in the beginning.